### PR TITLE
test: Add e2e test for default value use cases for default stream and exclusive stream 

### PR DESCRIPTION
**Pull request by GaoleMeng**
Also fixed a bug in the testJsonStreamWriterWithDefaultStream, this was a time zone dependent test. We need to specify the timezone explicitly during assert.

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigquerystorage/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

If you write sample code, please follow the [samples format](
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md).

### CODE REVIEW

File: google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/it/ITBigQueryWriteManualClientTest.java [hunk 1]
```diff
@@ -718,6 +729,129 @@ public void testJsonStreamWriterWithDefaultStream()
...
+            "projects/%s/datasets/%s/tables/%s",
+            ServiceOptions.getDefaultProjectId(), DATASET, tableName);
+    tableInfo =
+        TableInfo.newBuilder(
+                TableId.of(DATASET, tableName),
+                StandardTableDefinition.of(
+                    Schema.of(
+                        com.google.cloud.bigquery.Field.newBuilder(
+                                "foo_with_default", LegacySQLTypeName.STRING)
+                            .setDefaultValueExpression("'default_value_for_test'")
```
Comment (yirutang) [inline google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/it/ITBigQueryWriteManualClientTest.java:748]: Let's add a current time test.

File: google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/it/ITBigQueryWriteManualClientTest.java [hunk 2]
```diff
@@ -718,6 +729,129 @@ public void testJsonStreamWriterWithDefaultStream()
...
+                            .setMode(Field.Mode.NULLABLE)
+                            .build(),
+                        com.google.cloud.bigquery.Field.newBuilder(
+                                "bar_without_default", LegacySQLTypeName.STRING)
+                            .setMode(Mode.NULLABLE)
+                            .build(),
+                        com.google.cloud.bigquery.Field.newBuilder(
+                                "date_with_default_to_current", LegacySQLTypeName.DATETIME)
+                            .setDefaultValueExpression("CURRENT_DATE()")
```
Comment (yirutang) [inline google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/it/ITBigQueryWriteManualClientTest.java:757]: CURRENT_TIMESTAMP? That is the most common usage.

File: google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/it/ITBigQueryWriteManualClientTest.java [hunk 3]
```diff
@@ -718,6 +729,129 @@ public void testJsonStreamWriterWithDefaultStream()
...
+
+    currentRow = iter.next();
+    assertEquals("default_value_for_test", currentRow.get(0).getStringValue());
+    assertEquals("a", currentRow.get(1).getStringValue());
+    LOG.warning("The string value is " + currentRow.get(2).getStringValue());
+    assertFalse(currentRow.get(2).getStringValue().isEmpty());
+
+    currentRow = iter.next();
+    assertEquals("default_value_for_test", currentRow.get(0).getStringValue());
+    assertFalse(currentRow.get(2).getStringValue().isEmpty());
```
Comment (yirutang) [inline google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/it/ITBigQueryWriteManualClientTest.java:849]: Could we verify the value is expected?

**Review by yirutang — APPROVED** [review]

File: google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/it/ITBigQueryWriteManualClientTest.java (hunk 3 above)
Comment (GaoleMeng) [reply google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/it/ITBigQueryWriteManualClientTest.java]: Done

File: google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/it/ITBigQueryWriteManualClientTest.java (hunk 2 above)
Comment (GaoleMeng) [reply google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/it/ITBigQueryWriteManualClientTest.java]: Done

**Pull request merged**